backend/src/estark/mod.rs

```diff
@@ -9,15 +9,32 @@ use std::{
     path::{Path, PathBuf},
 };
 
-use crate::{Backend, BackendFactory, Error, Proof};
+use crate::{Backend, BackendFactory, BackendOptions, Error, Proof};
 use powdr_ast::analyzed::Analyzed;
 
 use powdr_executor::witgen::WitgenCallback;
 use powdr_number::{buffered_write_file, write_polys_file, DegreeType, FieldElement};
 use serde::Serialize;
 use starky::types::{StarkStruct, Step, PIL};
 
-fn create_stark_struct(degree: DegreeType) -> StarkStruct {
+enum ProofType {
```

Comment:
I don't understand these proof types yet!

Reply:
It's similar to the halo2 case, but specific to what polygon/starky estark do to get a proof all the way to being verified on Ethereum:

- first proof is a GL STARK
- recursive proof that can take 1+ GL STARKS and verifies them in another GL STARK, so same as the previous one but has proofs as inputs
- recursive proof that can take a GL STARK and give a BN STARK
- recursive fflonk proof that takes a BN STARK and gives a BN SNARK

Reply:
Hmm, IDK, seems like very hard-coded way to encode this. What if I want to use a Halo2 proof instead of fflonk in the last step? It's also a bit redundant with the `--field` argument.

To do what you're describing I'd expect something like this:
```bash
# Initial proof
powdr prove ... --field gl --backend estark
# Compress a few times
powdr compress ... --field gl --backend estark
powdr compress ... --field gl --backend estark
powdr compress ... --field gl --backend estark
# Final STARK compression
powdr compress ... --field bn254 --backend estark
# Final SNARK
powdr compress ... --field bn254 --backend fflonk
```

But fine for this PR, we can figure out along the way what level of flexibility we really need!

Reply:
> What if I want to use a Halo2 proof instead of fflonk in the last step?

If you started with STARKS, you can't.

>  It's also a bit redundant with the --field argument.

This is CLI implementation detail. The field is important for the circuit, not the recursion part of the proof system.

> To do what you're describing I'd expect something like this:

Idk, I think I disagree here. I'm not sure `compress` should be a special command. It is also just a proof, but with different inputs.

Yea agree to do this later!